### PR TITLE
Add Sessions to Crashlytics

### PR DESCRIPTION
**Pull request by mrober**
Add Sessions to Crashlytics and populate an internal custom key. Crashlytics still needs to be updated to add a new field to the Report for this id.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -39,6 +39,7 @@ buildscript {
         classpath 'com.google.gms:google-services:4.3.15'
         classpath "com.ncorti.ktfmt.gradle:plugin:0.11.0"
         classpath 'com.google.firebase:firebase-appdistribution-gradle:3.0.2'
+        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.9.5'
```
Comment (visumickey) [inline build.gradle:42]: Why is this change in here? Is this relevant to this integration?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/CrashlyticsRegistrar.java [hunk 2]
```diff
@@ -24,20 +24,29 @@
...
+  }
+
   @Override
   public List<Component<?>> getComponents() {
     return Arrays.asList(
         Component.builder(FirebaseCrashlytics.class)
             .name(LIBRARY_NAME)
             .add(Dependency.required(FirebaseApp.class))
             .add(Dependency.required(FirebaseInstallationsApi.class))
+            .add(Dependency.required(FirebaseSessions.class))
```
Comment (visumickey) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/CrashlyticsRegistrar.java:49]: Just a curiosity question here: Is the order important here? If so, why are we above the NativeComponent?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/FirebaseCrashlytics.java [hunk 3]
```diff
@@ -176,6 +179,26 @@ public Void call() throws Exception {
           }
         });
 
+    // TODO(mrober): Replace with a real session implementation.
+    firebaseSessions.register(new SessionSubscriber() {
+      @Override
+      public void onSessionChanged(@NonNull SessionDetails sessionDetails) {
+        Logger.getLogger().d("onSessionChanged: " + sessionDetails);
+        core.setInternalKey("sessionId", sessionDetails.getSessionId());
```
Comment (visumickey) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/FirebaseCrashlytics.java:187]: What does internalKey mean for Crashlytics? Is the plan to use this internal key to fill in the proto? If so, any implications in retaining this as a internalKey?

**Review by visumickey — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (mrober) [reply build.gradle:42]: Yes, otherwise any test app in this repo can't have Crashlytics.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/FirebaseCrashlytics.java (hunk 3 above)
Comment (mrober) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/FirebaseCrashlytics.java:188]: This is intended to be replaced with the new field in the report. I added a TODO for it.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/CrashlyticsRegistrar.java (hunk 2 above)
Comment (mrober) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/CrashlyticsRegistrar.java:49]: I don't think the order is important. I just grouped it with other Firebase* deps. Let me know if the order does matter and I can move it. But looking at the other deps, I don't see any real order.

**Review by visumickey — APPROVED** [review]
